sanitizers/src/main/java/com/code_intelligence/jazzer/sanitizers/RegexInjection.kt

```diff
@@ -33,34 +33,76 @@ object RegexInjection {
      */
     private const val CANON_EQ_ALMOST_EXPLOIT = "\u0300\u0300\u0300"
 
+    /**
+     * When injected into a regex pattern, helps the fuzzer break out of quotes and character
+     * classes in order to cause a [PatternSyntaxException].
+     */
+    private const val FORCE_PATTERN_SYNTAX_EXCEPTION_PATTERN = "\\E]\\E]]]]]]"
+
     // With CANON_EQ enabled, Pattern.compile allocates an array with a size that is
     // (super-)exponential in the number of consecutive Unicode combining marks. We use a mild case
     // of this as a magic string based on which we trigger a finding.
     // Note: The fuzzer might trigger an OutOfMemoryError or NegativeArraySizeException (if the size
     // of the array overflows an int) by chance before it correctly emits this "exploit". In that
     // case, we report the original exception instead.
-    @MethodHook(
-        type = HookType.BEFORE,
-        targetClassName = "java.util.regex.Pattern",
-        targetMethod = "compile",
-        targetMethodDescriptor = "(Ljava/lang/String;I)Ljava/util/regex/Pattern;"
+    @MethodHooks(
+        MethodHook(
```

Comment:
Should we also hook `String::replaceFirst`, `String::replaceAll`, `String::split` as those are probably often used.

Reply:
Thanks, I added hooks for them. An alternative would be to add `java.lang.String` to `additionalClassesToHook`, as it simply goes through the existing hooks on `Pattern.compile` and `Pattern.matches`. But I don't really feel comfortable opening `java.lang.String` up to all hooks...